pkg/ottl/contexts/internal/ottlcommon/map.go

```diff
@@ -15,26 +15,104 @@
 package ottlcommon // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/internal/ottlcommon"
 
 import (
+	"fmt"
+
 	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
 )
 
-func GetMapValue(attrs pcommon.Map, mapKey string) interface{} {
-	val, ok := attrs.Get(mapKey)
+func GetMapValue(m pcommon.Map, keys []ottl.Key) (interface{}, error) {
+	if len(keys) == 0 {
+		return nil, fmt.Errorf("cannot get map value without key")
+	}
+	if keys[0].String == nil {
+		return nil, fmt.Errorf("non-string indexing is not supported")
+	}
+
+	val, ok := m.Get(*keys[0].String)
 	if !ok {
-		return nil
+		return nil, nil
+	}
+	for i := 1; i < len(keys); i++ {
+		switch val.Type() {
+		case pcommon.ValueTypeMap:
+			if keys[i].String == nil {
+				return nil, fmt.Errorf("map must be indexed by a string")
+			}
+			val, ok = val.Map().Get(*keys[i].String)
+			if !ok {
+				return nil, fmt.Errorf("key not found in map")
+			}
+		case pcommon.ValueTypeSlice:
+			if keys[i].Int == nil {
+				return nil, fmt.Errorf("slice must be indexed by an int")
+			}
+			if int(*keys[i].Int) >= val.Slice().Len() {
```

Comment:
We should also check that the index is greater than 0 here.

Reply:
Done